techmatters/webchat
# Apply IP address block for Jamaica

## Description
Blocks the IP of the abusive contact as requested. This is done by attaching to the source code an empty list of blocked ips. At deploy time, we can bundle a new json that will override the empty list, to actually include blocked ips. This is done by fetching `blockedIps.json` from the same bucket as the webchat. I have already uploaded the blocklist to JM prod bucket.
NOTE: the `blockedIps.json` needs to have public access.

### Checklist
- [x] [Corresponding issue has been opened](https://bugs.benetech.org/browse/CHI-1356)

### Verification steps
- Get your ip address.
- Add your ip to `src/blockedIps.json`.
- `npm start` and check that the widget does not loads for you.
- If desired, upload the same json to the development webchat bucket (with the name `blockedIps.json`), deploy this branch, and check that the webchat does not appears for you. NOTE: the ip you get locally might differ than the one that we see in the Flex, so use the second one.
